src/figuresClasses.ts

```diff
@@ -1,11 +1,85 @@
-export interface Figure {}
+type Color = 'red' | 'green' | 'blue';
+type Shape = 'triangle' | 'circle' | 'rectangle';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+function roundToHundredths(value: number): number {
+  return Math.floor(value * 100) / 100;
+}
+
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('All sides must be greater than 0');
+    }
+
+    if (a + b <= c || a + c <= b || b + c <= a) {
+      throw new Error(
+        'The longest side must be less than the sum of the other two sides',
+      );
+    }
+  }
+
+  getArea(): number {
+    const s: number = (this.a + this.b + this.c) / 2;
+    const area: number = Math.sqrt(
+      s * (s - this.a) * (s - this.b) * (s - this.c),
+    );
+
+    return roundToHundredths(area);
+  }
+}
 
-export class Circle implements Figure {}
+export class Circle implements Figure {
+  shape: Shape = 'circle';
 
-export class Rectangle implements Figure {}
+  constructor(
```

Comment:
[CODE STYLE] When throwing an error, provide a more detailed description. For example: 'Circle radius must be greater than 0.'